mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSource.java

```diff
@@ -203,9 +203,11 @@ protected void validateField(FailureCollector collector, Schema.Field field, Sch
       // map datetimeoffset to timestamp which is invalid. In such case runtime will still fail even validation passes.
       // But we don't have the original source type information here and don't want to do big refactoring here
       if (actualFieldSchema.getLogicalType() == Schema.LogicalType.DATETIME &&
-            expectedFieldSchema.getLogicalType() == Schema.LogicalType.TIMESTAMP_MICROS ||
-            actualFieldSchema.getLogicalType() == Schema.LogicalType.DATETIME &&
-              expectedFieldSchema.getType() == Schema.Type.STRING) {
+        expectedFieldSchema.getLogicalType() == Schema.LogicalType.TIMESTAMP_MICROS ||
+        actualFieldSchema.getLogicalType() == Schema.LogicalType.DATETIME &&
+          expectedFieldSchema.getType() == Schema.Type.STRING ||
+        actualFieldSchema.getLogicalType() == Schema.LogicalType.TIMESTAMP_MICROS &&
+          expectedFieldSchema.getLogicalType() == Schema.LogicalType.DATETIME) {
```

Comment:
Create a new if block and add comments around to state what use case we are trying to cover.